synapse/metrics/__init__.py

```diff
@@ -139,12 +144,27 @@ def update_resource_metrics():
 }
 
 
+def update_resource_metrics():
+    global rusage
+    rusage = getrusage(RUSAGE_SELF)
+
+    if HAVE_PROC_SELF_STAT:
+        global stats
+        with open("/proc/self/stat") as s:
+            line = s.read()
+            # line is PID (command) more stats go here ...
+            stats = line.split(") ", 1)[1].split(" ")
```

Comment:
Can we parse the stats here, rather than requiring the users of stat to know the magic numbers?